guides/traversing-pages.rst

```diff
@@ -1,164 +1,200 @@
-but you also have access to `named selectors`_ and `css selectors`_. Mink
-will transform such selectors into XPath queries internally for you.
+Most usages of Mink will involve working with the page opened in your browser.
+This is done thanks to the powerful Element API. This API allows to traverse
+the page (similar to the DOM in Javascript) and to interact with it, which
+will be covered in the :doc:`next chapter </guides/manipulating-pages>`.
 
-The main class of Mink's selectors engine is ``Behat\Mink\Selector\SelectorsHandler``.
-It handles different selector types, which implements ``Behat\Mink\Selector\SelectorInterface``:
+DocumentElement and NodeElement
+-------------------------------
 
-.. code-block:: php
+The Element API consists of 2 main classes. The ``DocumentElement`` instance
+represents the page being displayed in the browser, while the ``NodeElement``
+class is used to represent any element inside the page. Both class are sharing
+a common set of methods to traverse the page (defined in ``TraversableElement``).
 
-    $cssSelector = new \Behat\Mink\Selector\CssSelector();
+The ``DocumentElement`` instance is accessible through the ``Session::getPage`` method:
 
-    // generate XPath query out of CSS:
-    echo $cssSelector->translateToXPath('h1 > a');
+.. code-block:: php
 
-    $handler = new \Behat\Mink\Selector\SelectorsHandler();
-    $handler->registerSelector('css', $cssSelector);
+    $page = $session->getPage();
 
-    // generate XPath query out of CSS:
-    echo $handler->selectorToXpath('css', 'h1 > a');
+    // You can now manipulate the page.
 
-When you initialize ``Selector\SelectorsHandler`` it already has `XPath selectors`_,
-`named selectors`_ and `css selectors`_ registered in it.
+.. note::
 
-You can provide a custom selectors handler as a second argument to your session
-instances:
+    The ``DocumentElement`` instance represents the ``<html>`` node in the
+    DOM. It is equivalent to ``document.documentElement`` in the Javascript
+    DOM API.
 
-.. code-block:: php
+Traversal Methods
+-----------------
 
-    $session = new \Behat\Mink\Session($driver,
-        new \Behat\Mink\Selector\SelectorsHandler()
-    );
+Elements have 2 main traversal methods: ``ElementInterface::findAll`` returns
+an array of ``NodeElement`` instances matching the provided :ref:`selector <selectors>`
+inside the current element while ``ElementInterface::find`` returns the first
+match or ``null`` when there is none.
 
-Mink will use this handler internally in `find* methods`_.
+The ``TraversableElement`` class also provides a bunch of shortcut methods
+on top of ``find()`` to make it easier to achieve many common use case:
 
-Named Selectors
-~~~~~~~~~~~~~~~
+``ElementInterface::has``
+    Checks whether a child element matches the given selector but without
+    returning it.
 
-Named selectors provide a way to get named XPath queries:
+``TraversableElement::findById``
+    Looks for a child element with the given id.
```

Comment:
I'm not sure if we should use `child` word here, because all `find` methods operate within element they are called on as mentioned above.